plugin/storage/es/spanstore/reader.go

```diff
@@ -153,6 +153,8 @@ func NewSpanReader(p SpanReaderParams) *SpanReader {
 		sourceFn:                      getSourceFn(p.Archive, p.MaxDocCount),
 		maxDocCount:                   p.MaxDocCount,
 		useReadWriteAliases:           p.UseReadWriteAliases,
+		logger:                        p.Logger,
+		tracer:                        p.Tracer,
```

Comment:
tracer is not getting initialised, I think. 😕

Reply:
easy debugging trick:
```
if p.Tracer == nil {
     panic("nil tracer")
}
```
This gives you a stack trace showing the order of execution. If that does not explain where nil is coming from, then bubble this code up the stack.